Sources/XMLCoder/Encoder/XMLEncoderImplementation.swift

```diff
@@ -81,6 +89,23 @@ class XMLEncoderImplementation: Encoder {
         return KeyedEncodingContainer(container)
     }
 
+    public func choiceContainer<Key>(keyedBy _: Key.Type) -> KeyedEncodingContainer<Key> {
+        let topContainer: SharedBox<ChoiceBox>
+        if canEncodeNewValue {
+            // We haven't yet pushed a container at this level; do so here.
+            topContainer = storage.pushChoiceContainer()
+        } else {
+            guard let container = storage.lastContainer as? SharedBox<ChoiceBox> else {
+                preconditionFailure("Attempt to push new (single element) keyed encoding container when already previously encoded at this path.")
```

Comment:
Again, this looks like a copy-the-style-as-it-is job. Do you want me to tidy up the other instances as well or just stick to the new code?

Reply:
yes please, much appreciated!

Reply:
Is there a strict line length for the repo? The `.swiftlint.yml` declares violations at `150` and `200`. I've been sticking to 100 out of habit, but I can constrain these to something else if it is so desired.

Reply:
I've been trying to get some balance between changing the whole repository and maintaining readability. 80 doesn't work well when indentation is 4 spaces instead of 2, but I decided to stick with 4 to preserve the changes history. You can clean up as much as you don't find tedious. I think last time I tried to tighten the limit in `.swiftlint.yml` it required too many code changes and reformatting XML snippets in the unit tests, which I just decided to postpone, but never got back to it. Maybe if there too many changes of this type, it can be made in a separate PR.

Reply:
All seems fair to me! Probably worth waiting for the official [swift-format](https://github.com/apple/swift-format) to roll out to act on anything?

Reply:
sure, makes sense

Reply:
For now, I just touched the long `preconditionFailure` messages.